tests/Integrations/Custom/Autoloaded/CommonScenariosTest.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace DDTrace\Tests\Integrations\Custom\Autoloaded;
+
+use DDTrace\Tests\Common\SpanAssertion;
+use DDTrace\Tests\Common\WebFrameworkTestCase;
+use DDTrace\Tests\Frameworks\Util\Request\RequestSpec;
+
+final class CommonScenariosTest extends WebFrameworkTestCase
+{
+    protected static function getAppIndexScript()
+    {
+        return __DIR__ . '/../../../Frameworks/Custom/Version_Autoloaded/public/index.php';
+    }
+
+    protected static function getEnvs()
+    {
+        return [
+            'APP_NAME' => 'custom_autoloaded_app',
+        ];
+    }
+
+    /**
+     * @dataProvider provideSpecs
+     * @param RequestSpec $spec
+     * @param array $spanExpectations
+     * @throws \Exception
+     */
+    public function testScenario(RequestSpec $spec, array $spanExpectations)
+    {
+        $traces = $this->tracesFromWebRequest(function () use ($spec) {
+            $this->call($spec);
+        });
+
+        $this->assertExpectedSpans($this, $traces, $spanExpectations);
+    }
+
+    public function provideSpecs()
+    {
+        return $this->buildDataProvider(
+            [
+                'A simple GET request returning a string' => [
+                    SpanAssertion::build(
+                        'web.request',
+                        'web.request',
+                        'web',
+                        'web.request'
```

Comment:
Not required for this PR, but we should probably provide an hook to allow users to set proper resource names (depending on their routing mechanism). Otherwise they will see in the UI one and only one resource for each and every request :)

Reply:
I agree! Like you said, maybe not for this PR, but if we provided an out-of-the-box low-cardinality URL-to-resource-name converter, do you think that would be useful? For example, it would convert URL's from `/user?id=123` to `/user?id=?` or something like that. Thoughts? :)

Reply:
Let's discuss for the future as well, we may also think about some nice integration with the agent, which has all the relevant info to do that for all the language integrations :)